addons/mail_plugin/controllers/mail_plugin.py

```diff
@@ -53,6 +53,71 @@ def res_partner_enrich_and_create_company(self, partner_id):
 
         return response
 
+    @http.route('/mail_plugin/partner/enrich_and_update_company', type='json', auth='outlook', cors='*')
+    def res_partner_enrich_and_update_company(self, partner_id):
+        """
+        Enriches an existing company using IAP
+        """
+        partner = request.env['res.partner'].browse(partner_id)
+        if not partner.is_company:
+            return {'error': _('Contact must be a company')}
+
+        normalized_email = partner.email_normalized
+        if not normalized_email:
+            return {'error': _('Contact has no valid email')}
+
+        domain = tools.email_domain_extract(normalized_email)
+        iap_data = self._iap_enrich(domain)
+
+        if 'enrichment_info' in iap_data:
+            return {'enrichment_info': iap_data['enrichment_info'], 'company': self._get_company_data(partner)}
+
+        phone_numbers = iap_data.get('phone_numbers')
+
+        partner_values = {}
+
+        if not partner.phone and phone_numbers:
+            partner_values.update({'phone': phone_numbers[0]})
+
+        if not partner.iap_enrich_info:
+            partner_values.update({'iap_enrich_info': json.dumps(iap_data)})
+
+        if not partner.image_128:
+            logo_url = iap_data.get('logo')
+            if logo_url:
+                try:
+                    response = requests.get(logo_url, timeout=2)
+                    if response.ok:
+                        partner_values.update({'image_1920': base64.b64encode(response.content)})
+                except Exception as e:
+                    _logger.warning('Download of image for updated company %s failed, error %s', partner.name, e)
```

Comment:
Do we really need this logger ? Is it really something we want to find in logs ?

Reply:
Up for this comment ?

Reply:
Sorry forgot to click on the comment button

This logger is used in FHU's version, I don't really know for what purpose it's used, do you want me to check that?

If we decide to remove it here, I think that it would be a good idea to also remove it elsewhere where it's used.

Reply:
We will not do anything with this logger I think, maybe only when debugging / improving the feature. I propose to remove it here and simply "pass". Concerning existing one well we will remove them one day (maybe).

Reply:
Okay then, passing!